From 5523a1a5431170491c3cb32b115e326d24732767 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:39 -0400
Subject: [ethernet] qlcnic: Add driver logs in error path

Message-id: <1408626422-3941-116-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88659
O-Subject: [PATCH RHEL 7.1 115/138] qlcnic: Add driver logs in error path.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id c65762fc2fc8a85c3d8a2024c17815e1a0f5dff0

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index 54448b0..1b5d375 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -879,6 +879,9 @@ int qlcnic_83xx_alloc_mbx_args(struct qlcnic_cmd_args *mbx,
    return 0;
   }
  }
+
+ dev_err(&adapter->pdev->dev, "%s: Invalid mailbox command opcode 0x%x\n",
+  __func__, type);
  return -EINVAL;
 }
 
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
index ba20c72..34d2737 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
@@ -2181,6 +2181,8 @@ int qlcnic_83xx_configure_opmode(struct qlcnic_adapter *adapter)
   max_sds_rings = QLCNIC_MAX_SDS_RINGS;
   max_tx_rings = QLCNIC_MAX_TX_RINGS;
  } else {
+  dev_err(&adapter->pdev->dev, "%s: Invalid opmode %d\n",
+   __func__, ret);
   return -EIO;
  }
 
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c
index c1e11f5..304e247 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c
@@ -1027,8 +1027,11 @@ int qlcnic_config_port_mirroring(struct qlcnic_adapter *adapter, u8 id,
  u32 arg1;
 
  if (adapter->ahw->op_mode != QLCNIC_MGMT_FUNC ||
-     !(adapter->eswitch[id].flags & QLCNIC_SWITCH_ENABLE))
+     !(adapter->eswitch[id].flags & QLCNIC_SWITCH_ENABLE)) {
+  dev_err(&adapter->pdev->dev, "%s: Not a management function\n",
+   __func__);
   return err;
+ }
 
  arg1 = id | (enable_mirroring ? BIT_4 : 0);
  arg1 |= pci_func << 8;
@@ -1318,8 +1321,12 @@ int qlcnic_config_switch_port(struct qlcnic_adapter *adapter,
  u32 arg1, arg2 = 0;
  u8 pci_func;
 
- if (adapter->ahw->op_mode != QLCNIC_MGMT_FUNC)
+ if (adapter->ahw->op_mode != QLCNIC_MGMT_FUNC) {
+  dev_err(&adapter->pdev->dev, "%s: Not a management function\n",
+   __func__);
   return err;
+ }
+
  pci_func = esw_cfg->pci_func;
  index = qlcnic_is_valid_nic_func(adapter, pci_func);
  if (index < 0)
@@ -1363,6 +1370,8 @@ int qlcnic_config_switch_port(struct qlcnic_adapter *adapter,
    arg1 &= ~(0x0ffff << 16);
    break;
  default:
+  dev_err(&adapter->pdev->dev, "%s: Invalid opmode 0x%x\n",
+   __func__, esw_cfg->op_mode);
   return err;
  }
 
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index d73bc5e..2dbd77c 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -1011,6 +1011,8 @@ int qlcnic_init_pci_info(struct qlcnic_adapter *adapter)
 
   if (pfn >= ahw->max_vnic_func) {
    ret = QL_STATUS_INVALID_PARAM;
+   dev_err(&adapter->pdev->dev, "%s: Invalid function 0x%x, max 0x%x\n",
+    __func__, pfn, ahw->max_vnic_func);
    goto err_eswitch;
   }
 
@@ -2537,8 +2539,7 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
     dev_err(&pdev->dev, "Adapter initialization failed. Please reboot\n");
     goto err_out_free_hw;
    default:
-    dev_err(&pdev->dev, "Adapter initialization failed. A reboot may be required to recover from this failure\n");
-    dev_err(&pdev->dev, "If reboot does not help to recover from this failure, try a flash update of the adapter\n");
+    dev_err(&pdev->dev, "Adapter initialization failed. Driver will load in maintenance mode to recover the adapter using the application\n");
     goto err_out_maintenance_mode;
    }
   }
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c
index d04888b..e6c38b2 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c
@@ -362,6 +362,8 @@ int qlcnic_is_valid_nic_func(struct qlcnic_adapter *adapter, u8 pci_func)
   if (adapter->npars[i].pci_func == pci_func)
    return i;
  }
+
+ dev_err(&adapter->pdev->dev, "%s: Invalid nic function\n", __func__);
  return -EINVAL;
 }
 
-- 
1.7.1